src/ServiceProviders/RepositoryProviders/GitLab/GitLabProvider.php

```diff
@@ -0,0 +1,246 @@
+<?php
+
+namespace Pantheon\TerminusBuildTools\ServiceProviders\RepositoryProviders\GitLab;
+
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+
+use Pantheon\TerminusBuildTools\Credentials\CredentialClientInterface;
+use Pantheon\TerminusBuildTools\Credentials\CredentialProviderInterface;
+use Pantheon\TerminusBuildTools\ServiceProviders\RepositoryProviders\GitProvider;
+use Pantheon\TerminusBuildTools\Credentials\CredentialRequest;
+use Pantheon\TerminusBuildTools\Utility\ExecWithRedactionTrait;
+use Pantheon\TerminusBuildTools\ServiceProviders\RepositoryProviders\RepositoryEnvironment;
+use Robo\Common\ConfigAwareTrait;
+use Robo\Config\Config;
+
+/**
+ * Encapsulates access to GitLab through git and the GitLab API.
+ */
+class GitLabProvider implements GitProvider, LoggerAwareInterface, CredentialClientInterface
+{
+    use LoggerAwareTrait;
+    use ExecWithRedactionTrait;
+
+    const SERVICE_NAME = 'gitlab';
+    // We make this modifiable as individuals can self-host GitLab.
+    public $GITLAB_URL;
+    const GITLAB_TOKEN = 'GITLAB_TOKEN';
+    const GITLAB_CONFIG_PATH = 'command.build.provider.git.gitlab_url';
```

Comment:
There should definitely be common `const` for this. Perhaps it is sufficient for the CI provider to use the const of the Git provider.